File: android/app/src/main/java/com/mirego/cschat/controller/LoginController.java
package com.mirego.cschat.controller;

import com.mirego.cschat.models.LoginRequest;
import com.mirego.cschat.models.User;
import com.mirego.cschat.services.CSChatService;

import io.reactivex.Flowable;

public class LoginController {

    private final CSChatService chatService;

    public LoginController(CSChatService chatService) {
        this.chatService = chatService;
    }

    public Flowable<User> login(String username, String password) {
        return chatService.login(new LoginRequest(username));
    }

}

File: ios/CSChat/Classes/Domain/Controller/LoginController.swift
//
//  LoginController.swift
//  CSChat
//
//  Created by Hugo Lefrancois on 2017-03-23.
//  Copyright © 2017 Mirego. All rights reserved.
//

import UIKit

protocol LoginController
{
    var isLoggedObservable: Observable<Bool> { get }

    func login(username: String, password: String, completion: @escaping (_ success: Bool) -> (Void))
}
